src/Storages/ObjectStorage/Local/Configuration.cpp

```diff
@@ -30,6 +30,21 @@ void StorageLocalConfiguration::fromNamedCollection(const NamedCollection & coll
     paths = {path};
 }
 
+void StorageLocalConfiguration::fromDisk(const String & disk_name, ASTs & args, ContextPtr context, bool with_structure)
+{
+    auto disk = context->getDisk(disk_name);
+    ParseFromDiskResult parsing_result = parseFromDisk(args, with_structure, context);
+
+    fs::path root = disk->getPath();
+    fs::path suffix = parsing_result.path_suffix.value_or("");
```

Comment:
Let's normalize `path_suffix` and check that it does not go beyond `root` (with `../../`)